Magento_Theme/web/js/mixins/gallery-mixin.js

```diff
@@ -2,11 +2,107 @@ define([
   'jquery',
   'underscore',
   'matchMedia',
-], function ($, _, mediaCheck) {
+  'mage/template',
+  'text!mage/gallery/gallery.html',
+  'mage/translate'
+], function ($, _, mediaCheck, template, galleryTpl, $t) {
   'use strict';
 
   return function (targetModule) {
+    /**
+    * Retrieves index if the main item.
+    * @param {Array.<Object>} data - Set of gallery items.
+    */
+    var getMainImageIndex = function (data) {
+      var mainIndex;
+
+      if (_.every(data, function (item) {
+        return _.isObject(item);
+      })
+      ) {
+        mainIndex = _.findIndex(data, function (item) {
+          return item.isMain;
+        });
+      }
+
+      return mainIndex > 0 ? mainIndex : 0;
+    };
     return targetModule.extend({
+      /**
+      * Initializes gallery with configuration options.
+      */
+      initGallery: function () {
+        var breakpoints = {},
+          settings = this.settings,
+          config = this.config,
+          tpl = template(galleryTpl, {
+            next: $t('Next'),
+            previous: $t('Previous'),
+            exitFullScreen: $t('Exit fullscreen'),
+            zoomIn: $t('Zoom in'),
+            zoomOut: $t('Zoom out')
```

Comment:
please use names pointing that these are labels: `nextLabel`, `previousLabel` etc.

Reply:
done 👍